src/Symfony/Component/HttpClient/EventSourceHttpClient.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient;
+
+use Symfony\Component\HttpClient\Chunk\ServerSentEvent;
+use Symfony\Component\HttpClient\Exception\EventSourceException;
+use Symfony\Component\HttpClient\Response\AsyncContext;
+use Symfony\Component\HttpClient\Response\AsyncResponse;
+use Symfony\Contracts\HttpClient\ChunkInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * @author Antoine Bluchet <soyuka@gmail.com>
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+final class EventSourceHttpClient implements HttpClientInterface
+{
+    use AsyncDecoratorTrait;
+    use HttpClientTrait;
+
+    private $reconnectionTime;
+
+    public function __construct(HttpClientInterface $client = null, float $reconnectionTime = 10.0)
+    {
+        $this->client = $client ?: HttpClient::create();
+        $this->reconnectionTime = $reconnectionTime;
+    }
+
+    public function connect($method, string $url, array $options = []): ResponseInterface
+    {
+        return $this->request($method, $url, self::mergeDefaultOptions($options, [
+            'buffer' => false,
+            'headers' => [
+                'Accept' => 'text/event-stream',
+                'Cache-Control' => 'no-cache',
+            ],
+        ], true));
+    }
+
+    public function request(string $method, string $url, array $options = []): ResponseInterface
```

Comment:
I'm wondering whether it make sense for an EventSource implementation to still be a full HttpClient as well. Shouldn't it *use* a client inside it (a special one) instead ?

Also, will code using `->connect()` receive other chunks than event ones ?

Reply:
> Also, will code using ->connect() receive other chunks than event ones ?

Yes it does, full power to the end-user. As the API exposed here is rather low-level, having a full HttpClient makes things easier. Also, wrapping a special client inside a new class would add a rather useless class to this component in my opinion.

Reply:
Also, having this implement `HttpClientInterface` allows streaming both regular requests and event-stream at the same time.